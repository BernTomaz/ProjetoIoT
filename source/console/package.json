{
  "name": "iot-device-simulator-console",
  "version": "3.0.2",
  "private": true,
  "dependencies": {
    "@aws-amplify/api": "^4.0.6",
    "@aws-amplify/auth": "^4.1.2",
    "@aws-amplify/core": "^4.3.0",
    "@aws-amplify/geo": "^1.0.0",
    "@aws-amplify/interactions": "^4.0.6",
    "@aws-amplify/storage": "^4.3.1",
    "@aws-amplify/ui-react": "^1.2.26",
    "@maplibre/maplibre-gl-geocoder": "^1.5.0",
    "@testing-library/jest-dom": "^5.11.4",
    "@testing-library/react": "^12.1.2",
    "@testing-library/user-event": "^13.3.0",
    "@types/jest": "^27.0.2",
    "@types/node": "^16.11.0",
    "@types/react": "^17.0.30",
    "@types/react-dom": "^17.0.0",
    "@types/react-router-dom": "^5.3.3",
    "aws-sdk": "^2.1354.0",
    "bootstrap": "^4.6.0",
    "bootstrap-icons": "^1.6.0",
    "maplibre-gl": "^2.4.0",
    "maplibre-gl-js-amplify": "^3.0.5",
    "moment": "^2.29.1",
    "react": "~16.14.0",
    "react-bootstrap": "^1.6.4",
    "react-dom": "~16.14.0",
    "react-router-dom": "~5.3.4",
    "typescript": "^4.4.4",
    "web-vitals": "^2.1.2"
  },
  "devDependencies": {
    "react-scripts": "^5.0.1"
  },
  "overrides": {
    "css-select@2.1.0": {
      "nth-check": "2.1.1"
    },
    "fast-xml-parser": "4.2.4"
  },
  "scripts": {
    "start": "react-scripts start",
    "build": "GENERATE_SOURCEMAP=false INLINE_RUNTIME_CHUNK=false react-scripts build",
    "test": "react-scripts test",
    "eject": "react-scripts eject"
  },
  "engines": {
    "node": ">=18.0.0"
  },
  "eslintConfig": {
    "extends": [
      "react-app",
      "react-app/jest"
    ]
  },
  "browserslist": {
    "production": [
      ">0.2%",
      "not dead",
      "not op_mini all"
    ],
    "development": [
      "last 1 chrome version",
      "last 1 firefox version",
      "last 1 safari version"
    ]
  }
}
