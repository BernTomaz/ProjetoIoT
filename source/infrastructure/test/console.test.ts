// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0

import "@aws-cdk/assert/jest";
import { SynthUtils } from "@aws-cdk/assert";
import { ConsoleConstruct } from "../lib/console";
import { Stack } from "aws-cdk-lib";
import { Bucket } from "aws-cdk-lib/aws-s3";

test('IoT Device Simulator ConsoleConstruct Test', () => {
    const stack = new Stack();
    const testBucket = new Bucket(stack, "testBucket", {});

    const console = new ConsoleConstruct(stack, 'TestCommonResource', {
        apiId: '12ab34cde5',
        s3LogsBucket: testBucket,
        adminEmail: "someEmail",
        mapArn: "arn:aws:geo:region:accountID:map/ExampleMap",
        placeIndexArn: "arn:aws:geo:region:accountID:place-index/ExamplePlaceIndex"
    });

    expect(SynthUtils.toCloudFormation(stack)).toMatchSnapshot();
});
