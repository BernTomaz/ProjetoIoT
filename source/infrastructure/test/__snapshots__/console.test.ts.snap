// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IoT Device Simulator ConsoleConstruct Test 1`] = `
{
  "Resources": {
    "TestCommonResourceDistributionCloudFrontDistributionCF3F0A1C": {
      "Metadata": {
        "cfn_nag": {
          "rules_to_suppress": [
            {
              "id": "W70",
              "reason": "Since the distribution uses the CloudFront domain name, CloudFront automatically sets the security policy to TLSv1 regardless of the value of MinimumProtocolVersion",
            },
          ],
        },
      },
      "Properties": {
        "DistributionConfig": {
          "Comment": "IoT Device Simulator Distribution",
          "CustomErrorResponses": [
            {
              "ErrorCode": 403,
              "ResponseCode": 200,
              "ResponsePagePath": "/index.html",
            },
            {
              "ErrorCode": 404,
              "ResponseCode": 200,
              "ResponsePagePath": "/index.html",
            },
          ],
          "DefaultCacheBehavior": {
            "CachePolicyId": "658327ea-f89d-4fab-a63d-7e88639e58f6",
            "Compress": true,
            "TargetOriginId": "TestCommonResourceDistributionCloudFrontDistributionOrigin17ECA9ED9",
            "ViewerProtocolPolicy": "redirect-to-https",
          },
          "DefaultRootObject": "index.html",
          "Enabled": true,
          "HttpVersion": "http2",
          "IPV6Enabled": true,
          "Logging": {
            "Bucket": {
              "Fn::GetAtt": [
                "testBucketDF4D7D1A",
                "RegionalDomainName",
              ],
            },
            "Prefix": "console-cf/",
          },
          "Origins": [
            {
              "DomainName": {
                "Fn::GetAtt": [
                  "TestCommonResourceDistributionS3Bucket4473F8A7",
                  "RegionalDomainName",
                ],
              },
              "Id": "TestCommonResourceDistributionCloudFrontDistributionOrigin17ECA9ED9",
              "S3OriginConfig": {
                "OriginAccessIdentity": {
                  "Fn::Join": [
                    "",
                    [
                      "origin-access-identity/cloudfront/",
                      {
                        "Ref": "TestCommonResourceDistributionCloudFrontDistributionOrigin1S3Origin537D11B7",
                      },
                    ],
                  ],
                },
              },
            },
          ],
        },
      },
      "Type": "AWS::CloudFront::Distribution",
    },
    "TestCommonResourceDistributionCloudFrontDistributionOrigin1S3Origin537D11B7": {
      "Properties": {
        "CloudFrontOriginAccessIdentityConfig": {
          "Comment": "Identity for TestCommonResourceDistributionCloudFrontDistributionOrigin17ECA9ED9",
        },
      },
      "Type": "AWS::CloudFront::CloudFrontOriginAccessIdentity",
    },
    "TestCommonResourceDistributionS3Bucket4473F8A7": {
      "DeletionPolicy": "Retain",
      "Properties": {
        "BucketEncryption": {
          "ServerSideEncryptionConfiguration": [
            {
              "ServerSideEncryptionByDefault": {
                "SSEAlgorithm": "AES256",
              },
            },
          ],
        },
        "LifecycleConfiguration": {
          "Rules": [
            {
              "NoncurrentVersionTransitions": [
                {
                  "StorageClass": "GLACIER",
                  "TransitionInDays": 90,
                },
              ],
              "Status": "Enabled",
            },
          ],
        },
        "LoggingConfiguration": {
          "DestinationBucketName": {
            "Ref": "testBucketDF4D7D1A",
          },
          "LogFilePrefix": "console-s3/",
        },
        "PublicAccessBlockConfiguration": {
          "BlockPublicAcls": true,
          "BlockPublicPolicy": true,
          "IgnorePublicAcls": true,
          "RestrictPublicBuckets": true,
        },
        "VersioningConfiguration": {
          "Status": "Enabled",
        },
      },
      "Type": "AWS::S3::Bucket",
      "UpdateReplacePolicy": "Retain",
    },
    "TestCommonResourceDistributionS3BucketPolicyFD5D32C0": {
      "Metadata": {
        "cfn_nag": {
          "rules_to_suppress": [
            {
              "id": "F16",
              "reason": "Public website bucket policy requires a wildcard principal",
            },
          ],
        },
      },
      "Properties": {
        "Bucket": {
          "Ref": "TestCommonResourceDistributionS3Bucket4473F8A7",
        },
        "PolicyDocument": {
          "Statement": [
            {
              "Action": "s3:*",
              "Condition": {
                "Bool": {
                  "aws:SecureTransport": "false",
                },
              },
              "Effect": "Deny",
              "Principal": {
                "AWS": "*",
              },
              "Resource": [
                {
                  "Fn::GetAtt": [
                    "TestCommonResourceDistributionS3Bucket4473F8A7",
                    "Arn",
                  ],
                },
                {
                  "Fn::Join": [
                    "",
                    [
                      {
                        "Fn::GetAtt": [
                          "TestCommonResourceDistributionS3Bucket4473F8A7",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
            {
              "Action": "s3:GetObject",
              "Effect": "Allow",
              "Principal": {
                "CanonicalUser": {
                  "Fn::GetAtt": [
                    "TestCommonResourceDistributionCloudFrontDistributionOrigin1S3Origin537D11B7",
                    "S3CanonicalUserId",
                  ],
                },
              },
              "Resource": {
                "Fn::Join": [
                  "",
                  [
                    {
                      "Fn::GetAtt": [
                        "TestCommonResourceDistributionS3Bucket4473F8A7",
                        "Arn",
                      ],
                    },
                    "/*",
                  ],
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::S3::BucketPolicy",
    },
    "TestCommonResourceIDSIoTPolicyB2F77C27": {
      "Metadata": {
        "cfn_nag": {
          "rules_to_suppress": [
            {
              "id": "W11",
              "reason": "Cannot specify the resource to attach policy to identity.",
            },
          ],
        },
      },
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": "iot:Connect",
              "Effect": "Allow",
              "Resource": {
                "Fn::Join": [
                  "",
                  [
                    "arn:",
                    {
                      "Ref": "AWS::Partition",
                    },
                    ":iot:",
                    {
                      "Ref": "AWS::Region",
                    },
                    ":",
                    {
                      "Ref": "AWS::AccountId",
                    },
                    ":client/*",
                  ],
                ],
              },
            },
            {
              "Action": "iot:Subscribe",
              "Effect": "Allow",
              "Resource": {
                "Fn::Join": [
                  "",
                  [
                    "arn:",
                    {
                      "Ref": "AWS::Partition",
                    },
                    ":iot:",
                    {
                      "Ref": "AWS::Region",
                    },
                    ":",
                    {
                      "Ref": "AWS::AccountId",
                    },
                    ":topicfilter/*",
                  ],
                ],
              },
            },
            {
              "Action": "iot:Receive",
              "Effect": "Allow",
              "Resource": {
                "Fn::Join": [
                  "",
                  [
                    "arn:",
                    {
                      "Ref": "AWS::Partition",
                    },
                    ":iot:",
                    {
                      "Ref": "AWS::Region",
                    },
                    ":",
                    {
                      "Ref": "AWS::AccountId",
                    },
                    ":topic/*",
                  ],
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IoT::Policy",
    },
    "TestCommonResourceIdentityPoolAuthenticatedRoleB174B008": {
      "Metadata": {
        "cfn_nag": {
          "rules_to_suppress": [
            {
              "id": "W11",
              "reason": "iot:AttachPrincipalPolicy does not allow for resource specification",
            },
          ],
        },
      },
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRoleWithWebIdentity",
              "Condition": {
                "ForAnyValue:StringLike": {
                  "cognito-identity.amazonaws.com:amr": "authenticated",
                },
                "StringEquals": {
                  "cognito-identity.amazonaws.com:aud": {
                    "Ref": "TestCommonResourceIdentityPoolFC1CEC73",
                  },
                },
              },
              "Effect": "Allow",
              "Principal": {
                "Federated": "cognito-identity.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "Description": {
          "Fn::Join": [
            "",
            [
              {
                "Ref": "AWS::StackName",
              },
              " Identity Pool authenticated role",
            ],
          ],
        },
        "Policies": [
          {
            "PolicyDocument": {
              "Statement": [
                {
                  "Action": "execute-api:Invoke",
                  "Effect": "Allow",
                  "Resource": {
                    "Fn::Join": [
                      "",
                      [
                        "arn:",
                        {
                          "Ref": "AWS::Partition",
                        },
                        ":execute-api:",
                        {
                          "Ref": "AWS::Region",
                        },
                        ":",
                        {
                          "Ref": "AWS::AccountId",
                        },
                        ":12ab34cde5/prod/*",
                      ],
                    ],
                  },
                },
              ],
              "Version": "2012-10-17",
            },
            "PolicyName": "ExecuteApiPolicy",
          },
          {
            "PolicyDocument": {
              "Statement": [
                {
                  "Action": [
                    "geo:SearchPlaceIndexForText",
                    "geo:GetMapGlyphs",
                    "geo:GetMapSprites",
                    "geo:GetMapStyleDescriptor",
                    "geo:SearchPlaceIndexForPosition",
                    "execute-api:Invoke",
                    "geo:GetMapTile",
                  ],
                  "Effect": "Allow",
                  "Resource": [
                    "arn:aws:geo:region:accountID:map/ExampleMap",
                    "arn:aws:geo:region:accountID:place-index/ExamplePlaceIndex",
                  ],
                },
              ],
              "Version": "2012-10-17",
            },
            "PolicyName": "LocationServicePolicy",
          },
          {
            "PolicyDocument": {
              "Statement": [
                {
                  "Action": "iot:AttachPrincipalPolicy",
                  "Effect": "Allow",
                  "Resource": "*",
                },
                {
                  "Action": "iot:Connect",
                  "Effect": "Allow",
                  "Resource": {
                    "Fn::Join": [
                      "",
                      [
                        "arn:",
                        {
                          "Ref": "AWS::Partition",
                        },
                        ":iot:",
                        {
                          "Ref": "AWS::Region",
                        },
                        ":",
                        {
                          "Ref": "AWS::AccountId",
                        },
                        ":client/*",
                      ],
                    ],
                  },
                },
                {
                  "Action": "iot:Subscribe",
                  "Effect": "Allow",
                  "Resource": {
                    "Fn::Join": [
                      "",
                      [
                        "arn:",
                        {
                          "Ref": "AWS::Partition",
                        },
                        ":iot:",
                        {
                          "Ref": "AWS::Region",
                        },
                        ":",
                        {
                          "Ref": "AWS::AccountId",
                        },
                        ":topicfilter/*",
                      ],
                    ],
                  },
                },
                {
                  "Action": "iot:Receive",
                  "Effect": "Allow",
                  "Resource": {
                    "Fn::Join": [
                      "",
                      [
                        "arn:",
                        {
                          "Ref": "AWS::Partition",
                        },
                        ":iot:",
                        {
                          "Ref": "AWS::Region",
                        },
                        ":",
                        {
                          "Ref": "AWS::AccountId",
                        },
                        ":topic/*",
                      ],
                    ],
                  },
                },
              ],
              "Version": "2012-10-17",
            },
            "PolicyName": "IoTPolicy",
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "TestCommonResourceIdentityPoolFC1CEC73": {
      "Properties": {
        "AllowUnauthenticatedIdentities": false,
        "CognitoIdentityProviders": [
          {
            "ClientId": {
              "Ref": "TestCommonResourceUserPoolClient2595A2FD",
            },
            "ProviderName": {
              "Fn::GetAtt": [
                "TestCommonResourceUserPoolAAE3EB23",
                "ProviderName",
              ],
            },
            "ServerSideTokenCheck": false,
          },
        ],
      },
      "Type": "AWS::Cognito::IdentityPool",
    },
    "TestCommonResourceIdentityPoolRoleAttachement3832B688": {
      "Properties": {
        "IdentityPoolId": {
          "Ref": "TestCommonResourceIdentityPoolFC1CEC73",
        },
        "Roles": {
          "authenticated": {
            "Fn::GetAtt": [
              "TestCommonResourceIdentityPoolAuthenticatedRoleB174B008",
              "Arn",
            ],
          },
        },
      },
      "Type": "AWS::Cognito::IdentityPoolRoleAttachment",
    },
    "TestCommonResourceUserEE889C83": {
      "Properties": {
        "DesiredDeliveryMediums": [
          "EMAIL",
        ],
        "ForceAliasCreation": true,
        "UserAttributes": [
          {
            "Name": "email",
            "Value": "someEmail",
          },
          {
            "Name": "email_verified",
            "Value": "true",
          },
        ],
        "UserPoolId": {
          "Ref": "TestCommonResourceUserPoolAAE3EB23",
        },
        "Username": "someEmail",
      },
      "Type": "AWS::Cognito::UserPoolUser",
    },
    "TestCommonResourceUserPoolAAE3EB23": {
      "DeletionPolicy": "Delete",
      "Properties": {
        "AccountRecoverySetting": {
          "RecoveryMechanisms": [
            {
              "Name": "verified_phone_number",
              "Priority": 1,
            },
            {
              "Name": "verified_email",
              "Priority": 2,
            },
          ],
        },
        "AdminCreateUserConfig": {
          "AllowAdminCreateUserOnly": true,
          "InviteMessageTemplate": {
            "EmailMessage": {
              "Fn::Join": [
                "",
                [
                  "
          <p>
            You are invited to join IoT Device Simulator.<br />
            https://",
                  {
                    "Fn::GetAtt": [
                      "TestCommonResourceDistributionCloudFrontDistributionCF3F0A1C",
                      "DomainName",
                    ],
                  },
                  "
          </p>
          <p>
            Please sign in to IoT Device Simulator using the temporary credentials below:<br />
            Username: <strong>{username}</strong><br />Password: <strong>{####}</strong>
          </p>
        ",
                ],
              ],
            },
            "EmailSubject": "[IoT Device Simulator] Login information",
          },
        },
        "AutoVerifiedAttributes": [
          "email",
        ],
        "EmailVerificationMessage": "The verification code to your new account is {####}",
        "EmailVerificationSubject": "Verify your new account",
        "Policies": {
          "PasswordPolicy": {
            "MinimumLength": 12,
            "RequireLowercase": true,
            "RequireNumbers": true,
            "RequireSymbols": true,
            "RequireUppercase": true,
          },
        },
        "SmsVerificationMessage": "The verification code to your new account is {####}",
        "UserPoolAddOns": {
          "AdvancedSecurityMode": "ENFORCED",
        },
        "UserPoolName": {
          "Fn::Join": [
            "",
            [
              {
                "Ref": "AWS::StackName",
              },
              "-user-pool",
            ],
          ],
        },
        "UsernameAttributes": [
          "email",
        ],
        "VerificationMessageTemplate": {
          "DefaultEmailOption": "CONFIRM_WITH_CODE",
          "EmailMessage": "The verification code to your new account is {####}",
          "EmailSubject": "Verify your new account",
          "SmsMessage": "The verification code to your new account is {####}",
        },
      },
      "Type": "AWS::Cognito::UserPool",
      "UpdateReplacePolicy": "Delete",
    },
    "TestCommonResourceUserPoolClient2595A2FD": {
      "Properties": {
        "AllowedOAuthFlows": [
          "implicit",
          "code",
        ],
        "AllowedOAuthFlowsUserPoolClient": true,
        "AllowedOAuthScopes": [
          "profile",
          "phone",
          "email",
          "openid",
          "aws.cognito.signin.user.admin",
        ],
        "CallbackURLs": [
          "https://example.com",
        ],
        "ClientName": {
          "Fn::Join": [
            "",
            [
              {
                "Ref": "AWS::StackName",
              },
              "-userpool-client",
            ],
          ],
        },
        "GenerateSecret": false,
        "PreventUserExistenceErrors": "ENABLED",
        "RefreshTokenValidity": 1440,
        "SupportedIdentityProviders": [
          "COGNITO",
        ],
        "TokenValidityUnits": {
          "RefreshToken": "minutes",
        },
        "UserPoolId": {
          "Ref": "TestCommonResourceUserPoolAAE3EB23",
        },
      },
      "Type": "AWS::Cognito::UserPoolClient",
    },
    "testBucketDF4D7D1A": {
      "DeletionPolicy": "Retain",
      "Properties": {
        "AccessControl": "LogDeliveryWrite",
      },
      "Type": "AWS::S3::Bucket",
      "UpdateReplacePolicy": "Retain",
    },
  },
}
`;
